src/data/finset/basic.lean

```diff
@@ -1504,6 +1504,231 @@ theorem forall_mem_insert [d : decidable_eq α]
   (∀ x, x ∈ insert a s → p x) ↔ p a ∧ (∀ x, x ∈ s → p x) :=
 by simp only [mem_insert, or_imp_distrib, forall_and_distrib, forall_eq]
 
+lemma eq_or_ne {α : Type*} [D : decidable_eq α] {x y : α} : (x = y)∨ (x ≠ y) :=
```

Comment:
This is `decidable.em (x = y)` (or more commonly `by_cases x = y`)

Reply:
Is this worth having anyway because it translates `¬(x = y)` into `x ≠ y`?

Reply:
I very often write `rw ← ne.def at h` at the start of the second case (to be able to write `h.symm` for example), and changing `by_cases` sounds like a good idea. Of course this is just a quick impression and it might be that changing this causes problems for other cases.